packages/ui/src/components/DateInput/Context.tsx

```diff
@@ -0,0 +1,72 @@
+import { createContext } from 'react'
+import type { Dispatch, SetStateAction } from 'react'
+import { DAYS_DEFAULT, MONTHS_ARR_DEFAULT, MONTHS_DEFAULT } from './constants'
+
+export type ContextProps = {
+  showMonthYearPicker?: boolean
+  disabled: boolean
+  /**
+   * month to show on popup - NOT selectedValue
+   */
+  monthToShow: number
+  /**
+   * Year to show on popup - NOT selectedValue
+   */
+  yearToShow: number
+  excludeDates?: Date[]
+  setMonthToShow: Dispatch<SetStateAction<number>>
+  setYearToShow: Dispatch<SetStateAction<number>>
+  minDate?: Date | null
+  maxDate?: Date | null
+  // Calendar months names and short names
+  MONTHS: Record<string, string>
+  // Week days names and shortnames
+  DAYS: Record<string, string>
+  MONTHS_ARR: string[]
+  setValue: Dispatch<SetStateAction<Date | null>>
+  setInputValue: Dispatch<SetStateAction<string | undefined>>
+  value?: Date | null
+  range?: {
+    start: Date | null
+    end: Date | null
+  }
+  setRange?: Dispatch<
+    SetStateAction<{
+      start: Date | null
+      end: Date | null
+    }>
+  >
+  startDate?: Date | null
+  endDate?: Date | null
+  format?: (value?: Date) => string | undefined
+} & (
+  | {
+      selectsRange: true
+      onChange?: (
+        date: [Date | null, Date | null],
+        event: React.SyntheticEvent | undefined,
+      ) => void
+    }
+  | {
+      selectsRange: false
+      onChange?: (
+        date: Date | null,
+        event: React.SyntheticEvent | undefined,
+      ) => void
+    }
+)
+export const DateInputContext = createContext<ContextProps>({
+  showMonthYearPicker: false,
+  disabled: false,
+  setValue: () => null,
+  setInputValue: () => null,
+  monthToShow: 1,
+  yearToShow: 2000,
+  setMonthToShow: () => null,
+  setYearToShow: () => null,
+  excludeDates: [],
+  MONTHS: MONTHS_DEFAULT,
+  DAYS: DAYS_DEFAULT,
+  MONTHS_ARR: MONTHS_ARR_DEFAULT,
```

Comment:
Why do you pass constant to the provider? Those could be accessed without the provider directly importing it from `constants.ts`